rtrlib/lib/vector.h

```diff
@@ -14,45 +14,80 @@
 #include <stdlib.h>
 #include <string.h>
 
+/**
+ * @brief ASPA Object
+ * here the customer as number (CAS) authorizes a set of provider AS numbers
+ */
+struct aspas_record {
+	uint32_t customer_as_number;
+	size_t number_of_provider_as;
+	uint32_t *provider_as_numbers;
+};
+
 /**
  * @brief Struct which is similar in function to std::vector from c++ so a reallocating array
  */
-struct lrtr_vector {
+struct aspa_vector {
 	uint32_t size;
 	uint32_t capacity;
-	uint32_t *data;
+	struct aspas_record *data;
 };
 /**
  * creates an vector object
  * @param[vector_pointer] the pointer to the newly created pointer will be written to *vector_pointer
- * @result Valid pointer to an lrtr_vector struct
+ * @result Valid pointer to an aspa_vector struct
  * @result Null On error.
  */
-int create_vector(struct lrtr_vector **vector_pointer);
+int create_vector(struct aspa_vector **vector_pointer);
 
 /**
  * deletes the given vector
- * @param[vector] lrtr_vector which will be deleted
+ * @param[vector] aspa_vector which will be deleted
  * @result 0 On success.
  * @result -1 On error.
  */
-int delete_vector(struct lrtr_vector *vector);
+int delete_vector(struct aspa_vector *vector);
 
 /**
  * reallocates the vector to increase its size
- * @param[vector] lrtr_vector which should get an increase in size
+ * @param[vector] aspa_vector which should get an increase in size
  * @result 0 On success.
  * @result -1 On error.
  */
-int reallocate_vector(struct lrtr_vector *vector);
+int reallocate_vector(struct aspa_vector *vector);
+
+/**
+ * sorts the given list of aspas records by the customer as number in ascending order
+ * @param[vector] list of aspas records
+ * @param[size] size of the aspas records list
+ */
+void quicksort(struct aspas_record *vector, size_t size);
 
 /**
  * adds a new value too the end of the given vector
- * @param[vector] lrtr_vector into which the value will be inserted
+ * @param[vector] aspa_vector into which the value will be inserted
  * @param[value] uin32_t value which will be inserted
  * @result 0 On success.
  * @result -1 On error.
  */
-int add_element_vector(struct lrtr_vector *vector, uint32_t value);
+int add_element_vector(struct aspa_vector *vector, struct aspas_record value);
+
+/**
+ * deletes a given element from the vector
+ * @param[vector] aspa_vector from where the element should be removed
+ * @param[index] index of the element which should be removed
+ * @result 0 On success.
+ * @result -1 On error.
+ */
+int delete_element_vector(struct aspa_vector *vector, size_t index);
```

Comment:
rename to vector_delete_at (similar for the other vector functions: start name with vector_)